src/box/iproto.cc

```diff
@@ -2460,27 +2462,24 @@ iproto_send_listen_msg(struct evio_service *binary)
 int
 iproto_listen(const char *uri)
 {
-	struct evio_service binary;
-	memset(&binary, 0, sizeof(binary));
-
 	if (iproto_send_stop_msg() != 0)
 		return -1;
-
-	if (uri != NULL) {
-		/*
-		 * Please note, we bind socket in main thread, and then
-		 * listen this socket in all iproto threads! With this
-		 * implementation, we rely on the Linux kernel to distribute
-		 * incoming connections across iproto threads.
-		 */
-		if (evio_service_bind(&binary, uri) != 0)
-			return -1;
-		if (iproto_send_listen_msg(&binary) != 0)
-			return -1;
+	if (evio_service_is_active(&tx_binary))
+		evio_service_stop(&tx_binary);
+	if (uri == NULL) {
+		tx_binary.addr_len = 0;
```

Comment:
AFAIS, addr length nullification is already done in `evio_service_stop()` via `evio_service_detach()`.

Reply:
Fixed